src/cli/ops/kernels.c

```diff
@@ -42,27 +41,10 @@ bool cbm_command_list_kernels(int argc, char **argv)
         boot_manager_set_update_efi_vars(manager, update_efi_vars);
 
         if (root) {
-                autofree(char) *realp = NULL;
-
-                realp = realpath(root, NULL);
```

Comment:
I think this test (and more the error message) are still useful even without caring about image mode. Without this the use gets an error of cannot inspect root which is less clear than the path not existing message that they would currently see (same comment for the other removals of this code below).